### ISSUE TITLE
Camera distance app error WIN10

### ISSUE THREAD

**Issue by Alexlusk** [searayeah/dota-camera-distance#21]
![image](https://github.com/searayeah/dota-camera-distance/assets/143673283/5fc5e30e-8ac7-425e-a0d1-8c28c96b7355)

**Comment by Alexlusk** [comment]
Windows 10. This issue started after the recent update on Aug 30 2023

**Comment by gpkmod** [comment]
> Windows 10. Эта проблема возникла после недавнего обновления от 30 августа 2023 г.

После обновления клиента возникает такая же проблема и у меня. Пожалуйста доделай код. Я уже не могу играть с дистанцией 1200 где виден только мой герой на пол экрана.

**Comment by AgitoReiKen** [comment]
Current patch
Offset: 1ECCAA
Data: C9 48 89 44 24 20 41 B8 00 40 00 00 C7 44 24 37 00 00 96 44 48 8D 15 3B 84 52 03 48 8D 0D 7C C3
Where 00 00 96 44 is 1200.0f

**Comment by searayeah** [comment]
> Current patch Offset: 1ECCAA Data: C9 48 89 44 24 20 41 B8 00 40 00 00 C7 44 24 37 00 00 96 44 48 8D 15 3B 84 52 03 48 8D 0D 7C C3 Where 00 00 96 44 is 1200.0f

Hey, where did you get this? If you have not found it yourself and know some forum or auto-script to find the code, send me a link, please, I am really tired of brute-force searching it every patch.

**Comment by AgitoReiKen** [comment]
> 
Brute-forced exactly. But maybe its possible to use pattern searching here.
![изображение](https://github.com/searayeah/dota-camera-distance/assets/42333163/eff8a9f7-d641-481a-b27e-2975542b96b2)

**Comment by Alexlusk** [comment]
> > 
> 
> Brute-forced exactly. But maybe its possible to use pattern searching here. ![изображение](https://user-images.githubusercontent.com/42333163/264705918-eff8a9f7-d641-481a-b27e-2975542b96b2.png)

Any idea what numbers would be for 1600 or 1650 distance instead of default 00 00 96 44 ?

**Comment by AgitoReiKen** [comment]
> 
Это обычный float в хексе
Онлайн Конвертер:
https://gregstoll.com/~gregstoll/floattohex/
![изображение](https://github.com/searayeah/dota-camera-distance/assets/42333163/d9c94fab-1c53-4448-b289-40f38fb90c2c)
т.е. 00 00 C8 44

**Comment by Alexlusk** [comment]
Спасибо братан

**Comment by AgitoReiKen** [comment]
> > Current patch Offset: 1ECCAA Data: C9 48 89 44 24 20 41 B8 00 40 00 00 C7 44 24 37 00 00 96 44 48 8D 15 3B 84 52 03 48 8D 0D 7C C3 Where 00 00 96 44 is 1200.0f
> 
> Hey, where did you get this? If you have not found it yourself and know some forum or auto-script to find the code, send me a link, please, I am really tired of brute-force searching it every patch.

I've made a script that uses byte-pattern matching. It's working atm.
Feel free to use it :)
https://github.com/AgitoReiKen/dota2cameradistance/blob/main/set_camera_distance.py

**Comment by searayeah** [comment]
> > > 
> > 
> > 
> > Brute-forced exactly. But maybe its possible to use pattern searching here. ![изображение](https://user-images.githubusercontent.com/42333163/264705918-eff8a9f7-d641-481a-b27e-2975542b96b2.png)
> 
> Any idea what numbers would be for 1600 or 1650 distance instead of default 00 00 96 44 ?

I've fixed the app yesterday, everything should work now

**Comment by searayeah** [comment]
> > > Current patch Offset: 1ECCAA Data: C9 48 89 44 24 20 41 B8 00 40 00 00 C7 44 24 37 00 00 96 44 48 8D 15 3B 84 52 03 48 8D 0D 7C C3 Where 00 00 96 44 is 1200.0f
> > 
> > 
> > Hey, where did you get this? If you have not found it yourself and know some forum or auto-script to find the code, send me a link, please, I am really tired of brute-force searching it every patch.
> 
> I've made a script that uses byte-pattern matching. It's working atm. Feel free to use it :) https://github.com/AgitoReiKen/dota2cameradistance/blob/main/set_camera_distance.py

Your script seems to work exactly like my app. I do the same pattern search and the pattern is located and updated from this file. Btw, length 16 is enough to unambiguously identify the code. https://github.com/searayeah/dota-camera-distance/blob/c0b7c71f039e36e94b0ce27346c7470c0eb8e137/current_hex_string#L1

The problem is updating this pattern on every patch as the byte surroundings of `00 00 96 44` are always different. See the codes for previous patches:

- 00 00 00 00 00 00 2E 40 00 00 96 44 00 00 E1 44
- 00 00 96 44 00 40 9C 44 00 80 A2 44 00 80 B4 44
- 00 00 AE 42 00 00 96 44 00 00 C8 44 00 40 9C 45

There is a way to automatically find it regardless of the patch and without using debuggers every time, but you need to know the structure of `.dll` files and how to calculate offset given a virtual memory address or a relative virtual address. I do not understand any of this at all.

**Comment by ev1lares** [comment]
![image](https://github.com/searayeah/dota-camera-distance/assets/143787463/15afb2f7-a1c9-4bc0-b91f-f8c3a605243d)


Please advise how do I fix thix? Release 5.0

**Comment by searayeah** [comment]
> ![image](https://user-images.githubusercontent.com/143787463/264953684-15afb2f7-a1c9-4bc0-b91f-f8c3a605243d.png)
> 
> Please advise how do I fix thix? Release 5.0

Relaunch the app, everything should work now.

**Comment by searayeah** [comment]
Apparently they changed the code twice in a 48h span.

**Comment by ev1lares** [comment]
> Apparently they changed the code twice in a 48h span.

Thanks man, it works.

**Comment by searayeah** [comment]
> > Apparently they changed the code twice in a 48h span.
> 
> Thanks man, it works. Is there any way to remove the artifacts on the top of the screen and remove the fog? I mean when the value is too high like 1500+

There should be a way, but it would be 100% cheating. I think some cheats support such thing. I have no time and do not want to implement this.

**Comment by ev1lares** [comment]
> > > Apparently they changed the code twice in a 48h span.
> > 
> > 
> > Thanks man, it works. Is there any way to remove the artifacts on the top of the screen and remove the fog? I mean when the value is too high like 1500+
> 
> There should be a way, but it would be 100% cheating. I think some cheats support such thing. I have no time and do not want to implement this.

Correct, I saw that features in some soft. Anyway, thanks at least for this app

**Comment by AgitoReiKen** [comment]
> > > > Current patch Offset: 1ECCAA Data: C9 48 89 44 24 20 41 B8 00 40 00 00 C7 44 24 37 00 00 96 44 48 8D 15 3B 84 52 03 48 8D 0D 7C C3 Where 00 00 96 44 is 1200.0f
> > > 
> > > 
> > > Hey, where did you get this? If you have not found it yourself and know some forum or auto-script to find the code, send me a link, please, I am really tired of brute-force searching it every patch.
> > 
> > 
> > I've made a script that uses byte-pattern matching. It's working atm. Feel free to use it :) https://github.com/AgitoReiKen/dota2cameradistance/blob/main/set_camera_distance.py
> 
> Your script seems to work exactly like my app. I do the same pattern search and the pattern is located and updated from this file. Btw, length 16 is enough to unambiguously identify the code.
> 
> https://github.com/searayeah/dota-camera-distance/blob/c0b7c71f039e36e94b0ce27346c7470c0eb8e137/current_hex_string#L1
> 
> The problem is updating this pattern on every patch as the byte surroundings of `00 00 96 44` are always different. See the codes for previous patches:
> 
>     * 00 00 00 00 00 00 2E 40 00 00 96 44 00 00 E1 44
> 
>     * 00 00 96 44 00 40 9C 44 00 80 A2 44 00 80 B4 44
> 
>     * 00 00 AE 42 00 00 96 44 00 00 C8 44 00 40 9C 45
> 
> 
> There is a way to automatically find it regardless of the patch and without using debuggers every time, but you need to know the structure of `.dll` files and how to calculate offset given a virtual memory address or a relative virtual address. I do not understand any of this at all.

You are doing it wrong because you are trying to find something that is getting changed everytime .dll gets compiled.
Check the pattern.txt in my repo, you will find a {4,4} things. They are used to identify 32 bit addresess used in instructions which are getting "randomized" each compilation.

**Comment by searayeah** [comment]
> > > > > Current patch Offset: 1ECCAA Data: C9 48 89 44 24 20 41 B8 00 40 00 00 C7 44 24 37 00 00 96 44 48 8D 15 3B 84 52 03 48 8D 0D 7C C3 Where 00 00 96 44 is 1200.0f
> > > > 
> > > > 
> > > > Hey, where did you get this? If you have not found it yourself and know some forum or auto-script to find the code, send me a link, please, I am really tired of brute-force searching it every patch.
> > > 
> > > 
> > > I've made a script that uses byte-pattern matching. It's working atm. Feel free to use it :) https://github.com/AgitoReiKen/dota2cameradistance/blob/main/set_camera_distance.py
> > 
> > 
> > Your script seems to work exactly like my app. I do the same pattern search and the pattern is located and updated from this file. Btw, length 16 is enough to unambiguously identify the code.
> > https://github.com/searayeah/dota-camera-distance/blob/c0b7c71f039e36e94b0ce27346c7470c0eb8e137/current_hex_string#L1
> > 
> > The problem is updating this pattern on every patch as the byte surroundings of `00 00 96 44` are always different. See the codes for previous patches:
> > ```
> > * 00 00 00 00 00 00 2E 40 00 00 96 44 00 00 E1 44
> > 
> > * 00 00 96 44 00 40 9C 44 00 80 A2 44 00 80 B4 44
> > 
> > * 00 00 AE 42 00 00 96 44 00 00 C8 44 00 40 9C 45
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > There is a way to automatically find it regardless of the patch and without using debuggers every time, but you need to know the structure of `.dll` files and how to calculate offset given a virtual memory address or a relative virtual address. I do not understand any of this at all.
> 
> You are doing it wrong because you are trying to find something that is getting changed everytime .dll gets compiled. Check the pattern.txt in my repo, you will find a {4,4} things. They are used to identify 32 bit addresess used in instructions which are getting "randomized" each compilation.


Oh, I opened the latest dll file in ida and it seems like they changed the logic and now `00 00 96 44` is "hardcoded" right after the opcode. Previously there was a memory address instead.

**Comment by Vladosbb** [comment]
they changed the code again?

**Comment by searayeah** [comment]
> they changed the code again?

Yep, they are constantly changing the file. Need to wait until I implement new logic which we discussed above.

**Comment by Vladosbb** [comment]
> > they changed the code again?
> 
> Yep, they are constantly changing the file. Need to wait until I implement new logic which we discussed above.
> 
> As for now, I manually updated the code, everything should work.

yeah, it works now, thank you :) It was easier to manually change before recent updates, now I don't even know what code to look for lol

**Comment by gpkmod** [comment]
> > > они снова поменяли код?
> > 
> > 
> > Да, они постоянно меняют файл. Нужно подождать, пока я реализую новую логику, о которой мы говорили выше.
> > На данный момент я обновил код вручную, все должно работать.
> 
> да, теперь работает, спасибо :) До недавних обновлений было проще вручную менять, теперь даже не знаю, какой код искать лол

Спасибо тебе огромное!

**Comment by searayeah** [comment]
> Опять не работает?

Download the latest version (6.1) with new implemented patten logic
